### ISSUE TITLE
SQL UPDATE SET fails when Strings contains quotes

### ISSUE THREAD

**Issue by mmacfadden** [orientechnologies/orientdb#4885]
SQL UPDATE SET statements are currently failing in 2.1.  

```Java
OCommandSQL command = new OCommandSQL("UPDATE test SET text = :text");
Map<String, Object> params = new HashMap<>();
params.put("text", "single \"");    
db.command(command).execute(params);
```

A string with a single quote produces an error: 
- "java.lang.IllegalStateException: Missing closed string character: '"'

A string with a quoted string in it produces the error: 
- Error on parsing command at position `#32`: Invalid keyword VALUE" STRING". Use UPDATE <class>|cluster:<cluster>> [SET|ADD|PUT|REMOVE|INCREMENT|CONTENT {<JSON>}|MERGE {<JSON>}] [[,] <field-name> = <expression>|<sub-command>]* [LOCK <NONE|RECORD>] [UPSERT] [RETURN <COUNT|BEFORE|AFTER>] [WHERE <conditions>]

I assume the issue is in the parser, but I haven't debugged much yet.  I have uploaded a unit test to a gist for review.

https://gist.github.com/mmacfadden/2be681114850ef30a6d0

**Comment by mmacfadden** [comment]
So far I have determined that the OUpdateStatement.replaceParameters method is calling OUpdateOperations.replaceParameters which is turning 

```
SET text = :text
```
into this:
```
SET text = "quoted "value" string"
```
The resulting update statement then looks like this:
```
UPDATE test SET text = "quoted "value" string"
```

which is problematic.  You wind up with SET text = "quoted" followed by what appears to be a keyword called 'value'.

I am still digging.

**Comment by mmacfadden** [comment]
It appears that the issue may lie in the OInputParameter.toParsedTree.  If the bound parameter is a string, it is just returns the string directly, where is probably needs to escape the quotes.  I am going to test out a fix in the morning and submit a PR if it works.

**Comment by mmacfadden** [comment]
Upon further investigation.  It appears that this is fixed in the current develop branch.  I am going to do a PR that just adds the unit tests to the OCommandExecutorSQLUpdateTest class.

**Comment by lvca** [comment]
@luigidellaquila Please could you merge this in 2.1.x branch so it will be available on the 2.1.1 release today?

**Comment by luigidellaquila** [comment]
yes, I'm merging it right now

**Comment by mmacfadden** [comment]
I believe it was a change to the ONamedParameter class that actually fixed this issue.  The toString method uses the OExpression.encode(String) method to escape the quotes.  This change was in the develop branch.  Was the change already merged into the 2.1.x branch?

**Comment by luigidellaquila** [comment]
Hi @mmacfadden 

I just fixed the problem in 2.1.x branch.
In develop branch it was fixed because I changed a bit query parsing process, anyway the final result now should be the same.

Thanks

**Comment by mmacfadden** [comment]
Great!
